gate-oauth2/src/test/java/com/netflix/spinnaker/gate/security/oauth2/SpinnakerUserInfoTokenServicesTest.java

```diff
@@ -100,6 +106,27 @@ public void shouldEvaluateUserInfoRequirementsAgainstAuthenticationDetails() {
         .isFalse();
   }
 
+  @Test
+  public void verifyIsServiceAccount() {
+    PermissionService permissionService = mock(PermissionService.class);
+    when(permissionService.isEnabled()).thenReturn(true);
+
+    Front50Service front50Service = mock(Front50Service.class);
+    ServiceAccount serviceAccount = new ServiceAccount();
+    serviceAccount = serviceAccount.setName("ex@foo.com");
+    when(front50Service.getServiceAccounts()).thenReturn(Calls.response(List.of(serviceAccount)));
+
+    SpinnakerUserInfoTokenServices tokenServices = new SpinnakerUserInfoTokenServices();
+    tokenServices.userInfoMapping = new OAuth2SsoConfig.UserInfoMapping();
+    tokenServices.permissionService = permissionService;
+    tokenServices.front50Service = front50Service;
+
+    Map<String, Object> details = Map.of("client_email", "ex@foo.com");
+    boolean isServiceAccount = tokenServices.isServiceAccount(details);
+
+    assertThat(isServiceAccount).isTrue();
```

Comment:
There is some verification of method calls in the groovy test that would be nice to preserve here, e.g.:
```
    verify(permissionService).isEnabled();
```
and
```
    verify(front50Service).getServiceAccounts());
```

Reply:
https://github.com/rahul-chekuri/gate/commit/254e9741ad6b166e913b724a4840f5b696106272